### PR TITLE
Fix: Empty state placeholder is not displayed in various widgets (#19446) [ED-7957]

### PR DESCRIPTION
**Pull request by hein-obox**

### CODE REVIEW

**Review by TzviRabinovitch — CHANGES_REQUESTED** [review]
tests?

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const { test, expect } = require( '@playwright/test' );
+const WpAdminPage = require( '../../../../../../../pages/wp-admin-page' );
+
+test( 'Empty widget placeholder sanity test', async ( { page }, testInfo ) => {
```
Comment (louiswol94) [inline tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:4]: @hein-obox Please change the title. Maybe:
"Check if empty placeholder is displayed in Image Carousel"

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js (hunk 1 above)
Comment (hein-obox) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js]: Changed

**Review by louiswol94 — DISMISSED** [review]

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const { test, expect } = require( '@playwright/test' );
+const WpAdminPage = require( '../../../../../../../pages/wp-admin-page' );
+
+test( 'Check if the empty placeholder is displayed in the Image Carousel', async ( { page }, testInfo ) => {
+	// Arrange.
+	const wpAdmin = new WpAdminPage( page, testInfo ),
+		editor = await wpAdmin.useElementorCleanPost();
+
+	// Act.
+	await editor.addWidget( 'image-carousel' );
```
Comment (louiswol94) [inline tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:10]: @hein-obox Fix indentation :-)

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const { test, expect } = require( '@playwright/test' );
```
Comment (louiswol94) [inline tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:1]: @hein-obox I think the filename needs to be more descriptive as well. You will see there is a file `hidden-widgets-on-search.test.js`. So maybe call yours `widget-empty-state.test.js`

Comment (udidol) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:1]: For short descriptions of test modules there is the `test.describe()` method

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	const wpAdmin = new WpAdminPage( page, testInfo ),
+		editor = await wpAdmin.useElementorCleanPost();
+
+	// Act.
+	await editor.addWidget( 'image-carousel' );
+
+    const emptyViewPlaceholder = editor.getPreviewFrame().locator( '.elementor-widget-image-carousel .elementor-widget-empty-icon' );
+
+	// Assert.
+    await expect( emptyViewPlaceholder ).toHaveCount( 1 );
```
Comment (udidol) [inline tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:15]: Why use `toHaveCount()`? You could use `toBeVisible()` also.

**Review by udidol — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js (hunk 2 above)
Comment (hein-obox) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:10]: @louiswol94 fixed

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js (hunk 3 above)
Comment (hein-obox) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:1]: @udidol @louiswol94 

Can we do a quick huddle to discuss this? That might help for future reference as well. 

@udidol I have noticed that in most tests the describe method is used to indicate the run command name?
@louiswol94 I have folllowed the file and folder structure from the exisiting JS files. If we follow that method, then I am not sure if I should change the filename.

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js (hunk 4 above)
Comment (hein-obox) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:15]: @udidol 

My logic: 
- The `widget.js` file checks if the height of a widget = 0 and if the widget is `:visible`. 
- If so, then the JS file will insert a placeholder. 
- The empty insert placeholder should get a placeholder, as it has the height = 0 and is visible.
- All that I do in my test is to check if the placeholder has been inserted in the empty Image Carousel widget.

In other words, I am only testing the outcome of the `onRender()` method inside `widget.js` code and not the conditions. I am not sure whether that is good practice or not.

Comment (udidol) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:15]: Ok I get that

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js (hunk 3 above)
Comment (udidol) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:1]: It's the run command name and a very short general description for a group of tests

Comment (udidol) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:1]: I think this is ok

**Review by udidol — DISMISSED** [review]

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const { test, expect } = require( '@playwright/test' );
+const WpAdminPage = require( '../../../../../../../pages/wp-admin-page' );
+
+test( 'Check if the empty placeholder is displayed in the Image Carousel', async ( { page }, testInfo ) => {
+	// Arrange.
+	const wpAdmin = new WpAdminPage( page, testInfo ),
+		editor = await wpAdmin.useElementorCleanPost();
```
Comment (TzviRabinovitch) [inline tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:7]: Can you add all the widgets that need to be verified?

**Review by TzviRabinovitch — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js (hunk 5 above)
Comment (hein-obox) [reply tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js:7]: @TzviRabinovitch 
That could be like 20 widgets.

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**